surprise/prediction_algorithms/algo_base.py

```diff
@@ -34,6 +34,7 @@ def __init__(self, **kwargs):
         self.sim_options = kwargs.get('sim_options', {})
         if 'user_based' not in self.sim_options:
             self.sim_options['user_based'] = True
+        self.verbose = bool(kwargs.get('verbose', True))
```

Comment:
`self.verbose = ...` should not be in AlgoBase, I think this would create conflict with other classes such as `SVD`.

It's probably best to set the `verbose` attribute in every single class that make use of it, right before calling `AlgoBase.__init__`.

Reply:
Ideally, __init__() should be called before any assignments. Do you think that there will be any interference otherwise?
Also, is it ok to put verbose at SymmetricAlgo level instead of the 4 different sub classes?

Reply:
>Ideally, init() should be called before any assignments.

Yes you're right, I was on my phone and thought there was some printing done in `__init__` but there's none so we can set `verbose` after calling it.

> is it ok to put verbose at SymmetricAlgo level

Sure why not